controllers/mover/syncthing/syncthing_test.go

```diff
@@ -474,7 +477,62 @@ var _ = Describe("When an RS specifies Syncthing", func() {
 		})
 
 		Context("Syncthing API is being used properly", func() {
-			// TODO: test syncthing API
+			When("Syncthing server does not exist", func() {
+				It("Doesn't work", func() {
+					res, err := mover.Synchronize(ctx)
+					Expect(err).ToNot(BeNil())
+					Expect(res).To(Equal(cMover.InProgress()))
+				})
+			})
+
+			When("Syncthing server exists", func() {
+				It("Works", func() {
+					s := SyncthingConfig{}
+					ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+						switch r.URL.Path {
+						case "/rest/config":
+							if r.Method == "GET" {
+								res := s
+								resBytes, _ := json.Marshal(res)
+								fmt.Fprintln(w, string(resBytes))
+							} else if r.Method == "PUT" {
+								s := SyncthingConfig{}
+								err := json.NewDecoder(r.Body).Decode(&s)
+								if err != nil {
+									http.Error(w, "Error decoding request body", http.StatusBadRequest)
+									return
+								}
+								Expect(s).To(Equal(*mover.syncthing.Config))
+							}
+							return
+						case "/rest/system/status":
+							res := SystemStatus{}
+							resBytes, _ := json.Marshal(res)
+							fmt.Fprintln(w, string(resBytes))
+							return
+						case "/rest/system/connections":
+							res := SystemConnections{}
+							resBytes, _ := json.Marshal(res)
+							fmt.Fprintln(w, string(resBytes))
+							return
+						default:
+							return
+						}
+					}))
+
+					defer ts.Close()
+
+					mover.syncthing.APIConfig.APIURL = ts.URL
+					mover.syncthing.APIConfig.Client = ts.Client()
+                                        mover.syncthing.APIConfig.APIKey = "test"
+
+					err := mover.syncthing.FetchLatestInfo()
+					Expect(err).To(BeNil())
+
```

Comment:
I would also throw in some tests here which validate our Syncthing struct persisted the config returned from the server.